src/main/java/com/trueffect/validation/PersonValidation.java

```diff
@@ -1,75 +1,105 @@
 import com.trueffect.tools.DataResourse.TypeIdentifier;
 import com.trueffect.tools.RegularExpression;
+import java.util.Date;
 import java.util.regex.Pattern;
+
 /*
  * @author santiago.mamani
  */
 public class PersonValidation {
-      
-    public static boolean isValidTypeIdentifier(String typeId){
-        TypeIdentifier typeIdentifier = null;
-        if (typeId.length()>0)
-           return typeId.equals(typeIdentifier.CI.name())||
-                  typeId.equals(typeIdentifier.PASS.name())||
-                  typeId.equals(typeIdentifier.NIT.name());
-        return false;
+
+    public static boolean isEmpty(String typeId) {
+        return typeId == null;
     }
-      public static boolean isValidIdentifier(String identifier) {
-        if (identifier.length() > 0) {
-            return Pattern.matches(RegularExpression.CI,identifier)||
-                   Pattern.matches(RegularExpression.PASS,identifier)||
-                   Pattern.matches(RegularExpression.NIT,identifier) ;
+
+    public static boolean isValidTypeIdentifier(String typeId) {
+        boolean res = true;
+        try {
+            TypeIdentifier typeIdentifier = TypeIdentifier.valueOf(typeId);
+        } catch (Exception e) {
+            res = false;
         }
-        return false;
+        return res;
     }
 
-    public static boolean isValidFirstName(String firstName, int size){
-         if(firstName.length()>0) 
-    return firstName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.FIRST_NAME,firstName);
-         return false;
+    public static boolean isValidIdentifier(String identifier) {
+        return Pattern.matches(RegularExpression.CI, identifier)
+                || Pattern.matches(RegularExpression.PASS, identifier)
+                || Pattern.matches(RegularExpression.NIT, identifier);
+
     }
-    public  static boolean isValidSize(String name, int size){
-     return name.length()>=0 && name.length()<= DataResourse.MAXIMUM_VALUES;
+
+    public static boolean isValidFirstName(String firstName) {
+        return Pattern.matches(RegularExpression.FIRST_NAME, firstName);
     }
-    
-   public static boolean isValidLastName(String lastName, int size){
-    if(lastName.length()>0) 
-        if(lastName.contains(" "))
-       return lastName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.LAST_NAME_TWO,lastName);
-        
-    return lastName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.LAST_NAME_ONE,lastName);
+
+    public static boolean isValidSize(String name, int size) {
+        return name.length() <= size;
     }
-  
-    public static boolean isValidGenre(String genre) {
-        Genre genreEnum= null;
-        if (genre != null) {
-            return genre.equals(genreEnum.M.name()) || genre.equals(genreEnum.F.name());
+
+    public static boolean isValidLastName(String lastName) {
+        if (lastName.contains(" ")) {
```

Comment:
you can use this StringUtils.isBlank() --> https://stackoverflow.com/questions/23419087/stringutils-isblank-vs-string-isempty